state/cleanup.go

```diff
@@ -142,6 +148,8 @@ func (st *State) Cleanup() (err error) {
 			err = st.cleanupModelsForDyingController(args)
 		case cleanupMachinesForDyingModel:
 			err = st.cleanupMachinesForDyingModel()
+		case cleanupUnitsForDyingModel:
```

Comment:
Does the order matter here? Either way might be worthwhile adding a comment as to why we are removing units after the machines...

Reply:
cleanup units for dying model and the machines sibling are mutually exclusive; one is for caas models, the other for iaas models